frontend/src/components/home/index.tsx

```diff
@@ -1,32 +1,64 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Route, Redirect, Switch } from 'react-router-dom';
+import toastr from 'toastr';
 import Header from '../header';
 import Employees from '../employees';
 import Rota from '../rota';
 import Sidebar from '../sidebar';
 import Summary from '../summary';
 import homeStyles from './home.module.scss';
 import PageNotFound from '../pageNotFound';
+import { get } from '../../service/apiService';
+import { useAuth0 } from '../../react-auth0-spa';
+import NotRegistered from '../notRegistered';
+import FullPageLoader from '../fullPageLoader';
 
-const Home = () => (
-  <div className={homeStyles.home}>
-    <Sidebar />
-    <div className={`${homeStyles.right} ${homeStyles.column}`}>
-      <Header />
-      <main className={homeStyles.main}>
-        <Switch>
-          <Route path="/employees" component={Employees} />
-          <Route path="/rota" component={Rota} />
-          <Route path="/summary" component={Summary} />
-          <Route exact path="/">
-            <Redirect to="/employees" />
-          </Route>
-          <Route path="/callback" />
-          <Route component={PageNotFound} />
-        </Switch>
-      </main>
-    </div>
-  </div>
-);
+const Home = () => {
+  const { getTokenSilently } = useAuth0();
+  const [role, setRole] = useState<null | string>(null);
+  useEffect(() => {
+    const getRole = async () => {
+      try {
+        const token = await getTokenSilently();
+        const response = await get('http://localhost:8080/role', token);
+        const json = await response.json();
+        setRole(json);
+      } catch (err) {
+        toastr.error('User not found');
+        setRole('NONE');
+      }
+    };
+    getRole();
+  }, [getTokenSilently]);
+
+  if (role === 'MANAGER' || role === 'USER') {
+    return (
+      <div className={homeStyles.home}>
+        <Sidebar />
+        <div className={`${homeStyles.right} ${homeStyles.column}`}>
+          <Header />
+          <main className={homeStyles.main}>
+            <Switch>
+              {role === 'MANAGER' && (
+                <Route path="/employees" component={Employees} />
```

Comment:
Just a thought, but what happens if an employee manually tries to navigate to '/employees' endpoint?
We probably want to display some sort of 'unauthorised' message.
One way of doing this would be to have
`component={role === Role.MANAGER ? Employees : Unauthorised}`

As a side note, I'm worried that having `role === Role.MANAGER` may not scale well if we want to add more roles in the future. It may be better to have some `Role -> boolean` function `hasViewPermissions(role)`.
We did something similar on the back end.

Reply:
Yeah definitely think this would be a better approach. I'll add some sort of basic unauthorised page.

And yes a method call would make more sense here. I've been told to look at react-router-guards for routing stuff too.

Reply:
Ideally unauthorized users should not know anything about the admin side of the app. I mean if unauthorized user navigates to `/employees` they can just be redirected to a default page (in the same way they should if they navigate to `/invalid-path`.

Can be done with a guard like @emab said.

Reply:
Currently they are navigated to a 404 style page, but not via router guards. It uses a router `switch` so if the page isn't recognised (which it won't be if they don't have the role) then it displays the page not found page. Is this suitable or is there a better way?